src/main/java/org/broadinstitute/consent/http/service/ElasticSearchService.java

```diff
@@ -278,7 +278,12 @@ public Response indexDataset(Dataset dataset, User user) throws IOException {
   }
 
   public Response indexDatasets(List<Dataset> datasets, User user) throws IOException {
-    List<DatasetTerm> datasetTerms = datasets.stream().map(this::toDatasetTerm).toList();
+    // Datasets in list context may not have their study populated, so we need to ensure that is
+    // true before trying to index them in ES.
+    List<DatasetTerm> datasetTerms = datasets.stream()
+        .map(d -> datasetDAO.findDatasetById(d.getDatasetId()))
```

Comment:
If there are many datasets, isn't this inefficient? Why not use `DatasetDAO.findDatasetsByIdList()` instead?

Reply:
It is inefficient, but that method doesn't populate study information. Adding study info to `findDatasetsByIdList()` significantly degrades the performance of the query.